image_encode.py

```diff
@@ -0,0 +1,30 @@
+"""
+For encoding images returned by the camera device.
+"""
+import cv2
+import numpy as np
+
+def encode_image_as_bytes(ext: str, image: "np.ndarray") -> "tuple[bool, bytes | None]" :
+    """
+    Encodes an image (np.ndarray as an RGB matrix) and returns its byte sequence.
+
+    Parameters
+    ----------
+    ext: Desired image encoding format (e.g., ".png").
+    image: Input image.
+
+    Returns
+    -------
+    tuple[bool, bytes]
+        The first parameter represents if the image encoding is successful.
+        - If it is not successful, the second parameter will be None.
+        - If it is successful, the second parameter will be the byte representation
+          of the encoded image.
+    """
+    result, encoded_image = cv2.imencode(ext, image)
```

Comment:
Add a TODO that this will be changed once the different formats are profiled and then selected.